### ISSUE TITLE
Unexpected TRUNCATE event

### ISSUE THREAD

**Issue by impero78** [aigan/tail-file#1]
While using the library to tail a fairly sizeable Wildfly server.log file for a period of 5-10 minutes, I have noticed that sometimes unpredictably throws TRUNCATE events. The unwanted consequence is that, after that,  the tailing resumes by default from the beginning of the file.

The issue is that the event should not even be thrown, as the log file itself is never truncated (in my system the log rotation happens only once a day at the end of the day, and never happened while I performed my tests).

I can see in the code that the statement that control if the TRUNCATE event is thrown is:

`if( stat.size < self.pos ) ..`

Printing on the console the file size and the position index values when the event is triggered, I see that for some reason the index has run ahead of the EOF:
```
stat.size: 838545772
self.pos : 838546007           (235 bytes ahead of stat.size)
```

**Comment by aigan** [comment]
What version of tail-file?
What operating system?
Single log file or multiple?
Is it a consistent error?
Could you run it with DEBUG? With bash; `DEBUG=tail ./myprogram`

**Comment by aigan** [comment]
I am considering a possible race condition when the fs watcher gets a change event triggering a read before the callback of the stat call. I'm working on a bugfix.

**Comment by aigan** [comment]
Hopefully fixed in v1.4.10.
I didn't try to find a way for testing exactly this, but made the changes and all existing tests pass.

**Comment by impero78** [comment]
Version: 1.4.7
Node: 12.3.0
OS: Ubuntu 14.04
Log: is a single log file that rotates every day.
The error does not occur consistently. I  am in the process of running over and over again the program in the attempt to reproduce the issue.

**Comment by aigan** [comment]
> The error does not occur consistently.

Thank you. It's probably the race condition I hopefully resolved with v1.4.10

**Comment by impero78** [comment]
Tested the fix it all day long today and I could not reproduce anymore. Thank you!
